core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableLimitSortRule.java

```diff
@@ -57,7 +57,7 @@ public interface Config extends RelRule.Config {
         ImmutableEnumerableLimitSortRule.Config.of()
             .withOperandSupplier(b0 ->
                 b0.operand(LogicalSort.class)
-                    .predicate(sort -> sort.fetch != null)
+                    .predicate(sort -> sort.fetch != null && !sort.getSortExps().isEmpty())
```

Comment:
The offset should not be ignore here, according to my understanding, there should also be a lack of corresponding test cases here. `sort.getSortExps()` should be replaced with `sort.collation.getFieldCollations()`.

Reply:
Done!